src/modules/savemanager/savemanager-service.js

```diff
@@ -428,7 +442,152 @@ angular.module('anol.savemanager')
+                var url = self.pollingUrl + '?layer=' + layerName;
+                $http.get(url).then(function(resp) {
+                  self.lastPollingResults[layerName] = resp.data;
+                  success_cb(resp.data, resp.status);
+                }, function(resp) {
+                  error_cb(resp.data, resp.status)
+                });
+              }, this.pollingInterval);
+            };
+
+            SaveManager.prototype.stopPolling = function(layerName) {
+              var interval = this.activePollingIntervals[layerName];
+              if (!interval) {
+                console.log(`Cannot stop polling. Polling for layer ${layerName} not started.`);
+                return;
+              }
+              $interval.cancel(interval);
+              this.activePollingIntervals[layerName] = undefined;
+            };
+
+            SaveManager.prototype.hasPollingChanges = function(layer) {
+              if (!layer) {
+                return false;
+              }
+              var layerName = layer.name;
+              var pollingResult = this.lastPollingResults[layerName];
+              if (!pollingResult) {
+                return false;
+              }
+
+              var features = layer.olLayer.getSource().getFeatures();
+              var removedFeatures = this.removedFeatures[layerName] || [];
+              // changed if polling contains at least one feature with newer timestamp
+              // changed if polling is missing at least one feature
+              var changed = features.some(function(feat) {
+                var pollingItem = pollingResult.find(function(pollingItem) {
+                  return pollingItem.id === feat.getId();
+                });
+                if (pollingItem) {
+                  var pollingModified = new Date(pollingItem.modified);
+                  var featureModified = new Date(feat.get('modified'));
+                  if (pollingModified > featureModified) {
+                    return true;
+                  }
+                } else if(feat.getId()) {
+                  // only consider already existing features
+                  return true;
+                }
+              });
+              if (changed) {
+                return true;
+              }
+              // changed if polling contains a feature that does not exist yet
+              // and is not in removedFeatures
+              changed = pollingResult.some(function(pollingItem) {
+                var feat = features.find(function(feature) {
+                  return feature.getId() === pollingItem.id;
+                });
+                if (!feat) {
+                  var localRemoved = removedFeatures.find(function(removedFeat) {
+                    return removedFeat.getId() === pollingItem.id;
+                  });
+                  return !localRemoved;
+                }
+                return false;
+              });
```

Comment:
note for future: whole logic changes a little with the style changes that keep the removed feature in the source.

```suggestion
              var features = layer.olLayer.getSource().getFeatures();
              var removedFeatures = this.removedFeatures[layerName] || [];
              // changed if polling contains at least one feature with newer timestamp
              // changed if polling is missing at least one feature
              var changed = features.some(function(feat) {
                var pollingItem = pollingResult.find(function(pollingItem) {
                  return pollingItem.id === feat.getId();
                });
                if (pollingItem) {
                  var pollingModified = new Date(pollingItem.modified);
                  var featureModified = new Date(feat.get('modified'));
                  if (pollingModified > featureModified) {
                    return true;
                  }
                } else if (feat.get('_state') !== 'new') {
                  // only consider already existing features
                  return true;
                }
              });
              if (changed) {
                return true;
              }
              // changed if polling contains a feature that does not exist yet
              // and is not in removedFeatures
              changed = pollingResult.some(function(pollingItem) {
                var feat = features.find(function(feature) {
                  return feature.getId() === pollingItem.id;
                });
                return feat && feat.get('_state') == 'removed';
              });
```